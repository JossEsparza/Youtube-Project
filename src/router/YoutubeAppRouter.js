import { useContext } from 'react';
import { Redirect } from 'react-router-dom';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { FavoritePage } from '../pages/FavoritePage';
import { HomePage } from '../pages/HomePage';
import VideoPage from '../pages/VideoPage';
import ToggleContext from '../state/ToggleContext';
import { Sidebar } from './../components/sidebar/Sidebar';
import './../index.css';

export const YoutubeAppRouter = () => {
  const { currentToggle } = useContext(ToggleContext);
  return (
    <Router>
      <div className="main-layout">
        {currentToggle && <Sidebar />}
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route path="/" component={VideoPage} />
          <Route exact path="/favorites" component={FavoritePage} />
          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  );
};
