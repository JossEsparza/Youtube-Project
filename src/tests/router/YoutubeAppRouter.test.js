// import React from 'react';
// import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { YoutubeAppRouter } from '../../router/YoutubeAppRouter';
import { mount } from 'enzyme';
// import { YoutubeAppRouter } from '../../router/YoutubeAppRouter';
// import { ToggleContext } from '../../state/ToggleContext';
// import { useReducer } from 'react';
import { ToggleContext } from '../../state/ToggleContext';

describe('Tests the routes', () => {
  test('should display the Home Page', () => {
    const history = createMemoryHistory({ initialEntries: ['/home'] });
    expect(history.location.pathname).toBe('/home');
  });

  test('should display the Favorites Page', () => {
    const history = createMemoryHistory({ initialEntries: ['/favorites'] });
    expect(history.location.pathname).toBe('/favorites');
  });

  test('should display the Video Page', () => {
    const video_id = '8d2323jp';
    const history = createMemoryHistory({ initialEntries: ['/' + video_id] });
    expect(history.location.pathname).toBe('/8d2323jp');
  });

  test('should be displayed correctly', () => {
    const globalReducer = {
      inputValue: 'changuitos',
      currentToggle: false,
      isEnabled: false,
    };
    const wrapper = mount(
      <ToggleContext.Provider
        value={{
          globalReducer,
        }}
      >
        <YoutubeAppRouter />
      </ToggleContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should be displayed correctly with currentToggle=true', () => {
    const globalReducer = {
      inputValue: 'changuitos',
      currentToggle: true,
      isEnabled: false,
    };
    const wrapper = mount(
      <ToggleContext.Provider
        value={{
          globalReducer,
        }}
      >
        <YoutubeAppRouter />
      </ToggleContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
