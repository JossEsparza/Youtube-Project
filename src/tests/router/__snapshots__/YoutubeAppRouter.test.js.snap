// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the routes should be displayed correctly 1`] = `ReactWrapper {}`;

exports[`Tests the routes should be displayed correctly with currentToggle=true 1`] = `ReactWrapper {}`;
